### ISSUE TITLE
Welcome to Hacktoberfest!

### ISSUE THREAD

**Issue by peternewman** [OpenLightingProject/ola#1669]
Happy Hacktober! :ghost:

If you'd like to participate in Hacktoberfest via the Open Lighting Architecture then have a look at some of our sets of issues here:
https://github.com/OpenLightingProject/ola/issues?q=is%3Aopen+is%3Aissue+label%3ADifficulty-Easy

You can also filter by language if you'd rather write in C++ or Python for example.

You can also take a look at our Help Wanted page for guidance and ideas:
https://www.openlighting.org/openlightingproject/get-involved/contributing/

If you fancy something more challenging, either try some of the medium or challenging difficulty issues or if you wanted to try something even bigger you could see some of our previous Google Summer of Code suggestions (or a part of them):
https://wiki.openlighting.org/index.php/OLP_SOC_Ideas_Page

If you haven't already, sign up for Hacktoberfest here - https://hacktoberfest.digitalocean.com/
:shirt: Make 4 pull requests during the month of October and you will receive a free t-shirt (or you have the option to plant a tree if you already have enough t-shirts )

**Comment by peternewman** [comment]
Although Hacktoberfest is over for 2020 I'm going to leave this open as it's still a good general introduction to how to get involved with OLA.

**Comment by peternewman** [comment]
We're back again for 2021, so why not join in?

**Comment by peternewman** [comment]
Hi @PushkarAdhyapak ,

That's great, however this issue is just a landing one for our broader Hacktoberfest involvement. Please take a look at some of the links in the first post for ways you can get involved (such as easy difficulty issues, or some other options). Or alternatively reply here with some indication of which language(s) you are most proficient in, whether you prefer front or back-end development etc and we can try and point you in the best direction.

**Comment by PushkarAdhyapak** [comment]
OK. I am  learning android development. Frontend issues can be managed by me.

**Comment by peternewman** [comment]
Okay, I guess that's the Kotlin stuff on your profile. So we don't have an Android app, the closest someone got to was trying to get OLA working on Android (but this was around 8 years ago):
https://wiki.openlighting.org/index.php/OLA_on_Android

You can also see some previous discussion on our mailing list too:
https://groups.google.com/g/open-lighting/search?q=android

Do you do HTML/JS frontend stuff too, or just Android?

While it would certainly be welcome, I suspect the majority of the Android related challenges would probably be a bit much for Hacktoberfest, but don't let that stop you!

**Comment by PushkarAdhyapak** [comment]
Ya I agree. Thanks a lot for your guidance

**Comment by peternewman** [comment]
We're back again for this year's Hacktoberfest if anyone wants to contribute this time...

**Comment by Logan-kwan** [comment]
hi, i would like to contribute. i know basic python

**Comment by peternewman** [comment]
> hi, i would like to contribute. i know basic python

Excellent @Logan-kwan . Do you use part of OLA already for anything?

Most of our easy Python stuff is in the RDM test server:
https://github.com/OpenLightingProject/ola/issues?q=is%3Aopen+label%3ALanguage-Python+label%3ADifficulty-Easy

Or there are some slightly more challenging bits also using Python:
https://github.com/OpenLightingProject/ola/issues?q=is%3Aopen+label%3ALanguage-Python

**Comment by Logan-kwan** [comment]
hi @peternewman, i havent use OLA before but i could learn the basics and apply them to my PR. would that be alright?

**Comment by peternewman** [comment]
Yeah that should work, I'm not trying to scare you off, but you might need to get your head round some of the terminology. Imagine if you'd never been in a car before and then you're trying to make the speedometer show the speed in metres per second, or make the handbrake more effective, you'll need to get to know what those terms mean and how to make the car move before your can test what you've done.

If you're interested in lighting or entertainment or something, then it might be something useful to learn anyway!

**Comment by Logan-kwan** [comment]
hmm ok... i may want to back off this one, but thank you for the opportunity and information!

**Comment by peternewman** [comment]
Sorry for the delay @Logan-kwan . You might like to take a look at https://github.com/codespell-project/codespell/issues/2536 which is another project I work on that's Python based with a much lower entry-point in terms of skills and complexity.

**Comment by Shruti-Kaushik** [comment]
@peternewman hey! I would like to contribute this year. I know html, CSS and basic JS and can work on VS code.

**Comment by peternewman** [comment]
Hi @Shruti-Kaushik ,

The Hacktoberfest windows itself has now closed (it's only open in October, hence the name), but we always welcome contributions and you could use the time now to get up to speed ready for the next Hacktoberfest or GSOC if we re-enter that (and you're eligible).

In terms of HTML/CSS/JS, here's our list of easy issues:
https://github.com/OpenLightingProject/ola/issues?q=is%3Aopen+label%3ADifficulty-Easy+label%3ALanguage-JS

In particular there is a load of stuff listed in here:
https://github.com/OpenLightingProject/ola/issues/891

Or perhaps most valuably if you're good with migrations, what to do should hopefully be fairly well documented elsewhere:
https://github.com/OpenLightingProject/ola/issues/1741

**Comment by kripton** [comment]
Do we leave this open so it's valid for all years or do we want a new "issue" for each year so it appears first in the list of issues?

**Comment by DaAwesomeP** [comment]
You can pin issues, so we could leave this one open and pin it during October: https://docs.github.com/en/issues/tracking-your-work-with-issues/pinning-an-issue-to-your-repository

On the other hand, finding the current list of things to help with way down in the issue comments may not be as intuitive.

**Comment by kripton** [comment]
Right, pinning is a great idea and we can modify the issue's first post to have a links of easy-for-starters-issues :+1:

**Comment by peternewman** [comment]
Come join us for Hacktoberfest 2023!

**Comment by devmaiter** [comment]
Hi! I'm looking to start contributing to OLA. Following @peternewman's suggestion in #1994, I've offered to pick up #1848 (ola_trigger `--config` command-line variables) as a first contribution — feedback on scope welcome there. If there are other areas where a new contributor would be more useful right now, happy to be pointed in that direction.

**Comment by zayeralam** [comment]
Hi @peternewman , I'm a 2nd-year CS student proficient in C++ and Python. I would love to solve this easy-difficulty issue as my first contribution to 'OLA'. Could you please guide me or assign this to me?..

**Comment by peternewman** [comment]
> Hi [@peternewman](https://github.com/peternewman) , I'm a 2nd-year CS student proficient in C++ and Python. I would love to solve this easy-difficulty issue as my first contribution to 'OLA'. Could you please guide me or assign this to me?..

Hi @zayeralam 

This issue isn't really an issue as such, it's just a signpost for people to find other relevant issues (which I think you've already done so from the looks of things).

See the rest of the first post, or this for starters:

> have a look at some of our sets of issues here: https://github.com/OpenLightingProject/ola/issues?q=is%3Aopen+is%3Aissue+label%3ADifficulty-Easy
